sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OnlineRecognizerConfig.java

```diff
@@ -1,138 +1,149 @@
+
+import java.util.Objects;
+
+public class OnlineRecognizerConfig {
+    private final FeatureConfig featConfig;
+    private final OnlineModelConfig modelConfig;
+    private final OnlineLMConfig lmConfig;
+
+    private final OnlineCtcFstDecoderConfig ctcFstDecoderConfig;
+    private final EndpointConfig endpointConfig;
+    private final HomophoneReplacerConfig hr;
+    private final boolean enableEndpoint;
+    private final String decodingMethod;
+    private final int maxActivePaths;
+    private final String hotwordsFile;
+    private final float hotwordsScore;
+    private final String ruleFsts;
+    private final String ruleFars;
+    private final float blankPenalty;
+
+    private OnlineRecognizerConfig(Builder builder) {
+        this.featConfig = builder.featConfig;
+        this.modelConfig = builder.modelConfig;
+        this.lmConfig = builder.lmConfig;
+        this.ctcFstDecoderConfig = builder.ctcFstDecoderConfig;
+        this.endpointConfig = builder.endpointConfig;
+        this.hr = builder.hr;
+        this.enableEndpoint = builder.enableEndpoint;
+        this.decodingMethod = builder.decodingMethod;
+        this.maxActivePaths = builder.maxActivePaths;
+        this.hotwordsFile = builder.hotwordsFile;
+        this.hotwordsScore = builder.hotwordsScore;
+        this.ruleFsts = builder.ruleFsts;
+        this.ruleFars = builder.ruleFars;
+        this.blankPenalty = builder.blankPenalty;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public OnlineModelConfig getModelConfig() {
+        return modelConfig;
+    }
+
+    public static class Builder {
+        private FeatureConfig featConfig = FeatureConfig.builder().build();
+        private OnlineModelConfig modelConfig = OnlineModelConfig.builder().build();
+        private OnlineLMConfig lmConfig = OnlineLMConfig.builder().build();
+        private OnlineCtcFstDecoderConfig ctcFstDecoderConfig = OnlineCtcFstDecoderConfig.builder().build();
+        private EndpointConfig endpointConfig = EndpointConfig.builder().build();
+        private HomophoneReplacerConfig hr = HomophoneReplacerConfig.builder().build();
+        private boolean enableEndpoint = true;
+        private String decodingMethod = "greedy_search";
+        private int maxActivePaths = 4;
+        private String hotwordsFile = "";
+        private float hotwordsScore = 1.5f;
+        private String ruleFsts = "";
+        private String ruleFars = "";
+        private float blankPenalty = 0.0f;
+
+        public OnlineRecognizerConfig build() {
+          if (maxActivePaths <= 0) {
+              throw new IllegalArgumentException("maxActivePaths must be > 0");
+          }
+          if (hotwordsScore < 0) {
+              throw new IllegalArgumentException("hotwordsScore must be >= 0");
+          }
+          if (blankPenalty < 0) {
+              throw new IllegalArgumentException("blankPenalty must be >= 0");
+          }
```

Comment:
You can remove the checks.

Please only `mv` the original files without any modifications.